src/cascadia/TerminalSettingsModel/Profile.h

```diff
@@ -57,83 +61,101 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         void GenerateGuidIfNecessary() noexcept;
         static guid GetGuidOrGenerateForJson(const Json::Value& json) noexcept;
 
-        bool HasGuid() const noexcept;

```

Comment:
Wait so we can get rid of the `HasGuid` and `optional<GUID>` now because `GETSET_SETTING` is already basically doing this for us?

Reply:
Correct. `GETSET_SETTING` introduces...
- `HasGuid()`
- `Guid()` (getter)
- `Guid(guid)` (setter)
- `ClearGuid()`
as well as the optional member and a helper function that does most of the work for the getter